copy/single.go

```diff
@@ -152,12 +161,12 @@ func (c *copier) copySingleImage(ctx context.Context, unparsedImage *image.Unpar
 		return copySingleImageResult{}, err
 	}
 
-	destRequiresOciEncryption := (isEncrypted(src) && ic.c.options.OciDecryptConfig != nil) || c.options.OciEncryptLayers != nil
+	destRequiresOciEncryption := (isEncrypted(src) && opt.options.OciDecryptConfig != nil) || opt.options.OciEncryptLayers != nil
```

Comment:
An example: Compare how `copy/encryption` accesses `ic.c.options.Oci*`. So now the two parts can get out of sync if a caller to `copySingleImage` passes options that differ from `ic.c.options`.